akka-projection-kafka/src/main/scala/akka/projection/kafka/KafkaSourceProvider.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.kafka
+
+import scala.concurrent.Future
+
+import akka.annotation.InternalApi
+import akka.kafka.AutoSubscription
+import akka.kafka.ConsumerSettings
+import akka.kafka.scaladsl.Consumer
+import akka.projection.internal.MergeableOffsets
+import akka.projection.internal.MergeableOffsets.OffsetRow
+import akka.projection.scaladsl.SourceProvider
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.TopicPartition
+
+object KafkaSourceProvider {
+  private val regexTp = """(.+)-(\d+)""".r
+
+  /**
+   * Create a [[SourceProvider]] that resumes from externally managed offsets
+   */
+  def apply[Key, ValueEnvelope](
+      settings: ConsumerSettings[Key, ValueEnvelope],
+      subscription: AutoSubscription): SourceProvider[MergeableOffsets.Offset, ConsumerRecord[Key, ValueEnvelope]] =
```

Comment:
do we need custom AutoSubscription or should we make it `topics: Set[String]`?

Reply:
:+1: